lib/Indexer/Model/Record/HasFlags.php

```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Phpactor\Indexer\Model\Record;
+
+interface HasFlags
```

Comment:
this can simply be a method on `ClassRecord` - the flags apply only to class record and could have a specific return type based on that information. If other things need "flags" they can be implemented separately.